sdk/spring/azure-spring-boot-test-aad/src/test/java/com/azure/test/aad/selenium/AADSeleniumITHelper.java

```diff
@@ -1,115 +1,72 @@
     }
 
-    public AADSeleniumITHelper(Class<?> appClass, Map<String, String> properties) throws InterruptedException {
+    public AADSeleniumITHelper(Class<?> appClass, Map<String, String> properties) {
         username = AAD_USER_NAME_1;
         password = AAD_USER_PASSWORD_1;
-        app = new AppRunner(appClass);
-        DEFAULT_PROPERTIES.forEach(app::property);
-        properties.forEach(app::property);
-
-        ChromeOptions options = new ChromeOptions();
-        options.addArguments("--headless");
-        options.addArguments("--incognito", "--no-sandbox", "--disable-dev-shm-usage");
-        this.driver = new ChromeDriver(options);
-
-        this.app.start();
-        login();
+        this.appClass = appClass;
+        this.properties = properties;
     }
 
-    private void login() throws InterruptedException {
-        WebDriverWait wait = new WebDriverWait(this.driver, 10);
+    public void login() {
         driver.get(app.root() + "oauth2/authorization/azure");
-        wait.until(presenceOfElementLocated(By.name("loginfmt"))).sendKeys(username + Keys.ENTER);
-        Thread.sleep(10000);
-        driver.findElement(By.name("passwd")).sendKeys(password + Keys.ENTER);
-        Thread.sleep(10000);
+        wait.until(ExpectedConditions.elementToBeClickable(By.name("loginfmt"))).sendKeys(username + Keys.ENTER);
+        wait.until(ExpectedConditions.elementToBeClickable(By.name("passwd"))).sendKeys(password + Keys.ENTER);
         driver.findElement(By.cssSelector("input[type='submit']")).click();
-        Thread.sleep(10000);
     }
 
-    public String httpGet(String endpoint) throws InterruptedException {
+    public String httpGet(String endpoint) {
         driver.get((app.root() + endpoint));
-        Thread.sleep(1000);
-        return driver.findElement(By.tagName("body")).getText();
+        return wait.until(presenceOfElementLocated(By.tagName("body"))).getText();
     }
 
-    public void logoutTest() throws InterruptedException {
-        WebDriverWait wait = new WebDriverWait(driver, 10);
+    public void logoutTest() {
         driver.get(app.root() + "logout");
-        wait.until(presenceOfElementLocated(By.cssSelector("button[type='submit']"))).click();
-        Thread.sleep(10000);
+        wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("button[type='submit']"))).click();
         String cssSelector = "div[data-test-id='" + username + "']";
-        driver.findElement(By.cssSelector(cssSelector)).click();
-        Thread.sleep(10000);
-        String id = driver.findElement(By.cssSelector("div[tabindex='0']")).getAttribute("data-test-id");
+        wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector(cssSelector))).click();
+        String id = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("div[tabindex='0']")))
+            .getAttribute("data-test-id");
         Assert.assertEquals(username, id);
     }
+
+    @Override
+    public void appInit() {
```

Comment:
I prefer to complete the app initialization and driver construction in the helper's constructor, instead of letting IT class doing it.